src/main/java/org/wise/portal/service/peergrouping/logic/impl/DifferentIdeasLogicServiceImpl.java

```diff
@@ -0,0 +1,121 @@
+      List<Annotation> annotations, Set<Workgroup> workgroupsNotInPeerGroup) {
+    return workgroupsNotInPeerGroup.stream()
+        .filter(workgroup -> hasWorkgroupAnnotation(workgroup, annotations)).count() >= 2;
+  }
+
+  Set<Workgroup> groupMembersUpToMaxMembership(Workgroup workgroup, PeerGrouping peerGrouping,
+      Set<Workgroup> possibleMembers) {
+    Set<Workgroup> members = new HashSet<Workgroup>();
+    members.add(workgroup);
+    possibleMembers.remove(workgroup);
+    TreeSet<WorkgroupWithDifferentIdeas> workgroupsWithDifferentIdeas = getWorkgroupsWithDifferentIdeas(
+        possibleMembers, workgroup, peerGrouping);
+    Iterator<WorkgroupWithDifferentIdeas> iterator = workgroupsWithDifferentIdeas.iterator();
+    while (members.size() < peerGrouping.getMaxMembershipCount()) {
+      members.add(iterator.next().getWorkgroup());
+    }
+    return members;
+  }
+
+  private TreeSet<WorkgroupWithDifferentIdeas> getWorkgroupsWithDifferentIdeas(
+      Set<Workgroup> possibleMembers, Workgroup workgroup, PeerGrouping peerGrouping) {
+    List<Annotation> annotations = getIdeaDetectedAnnotations(peerGrouping, workgroup.getPeriod());
+    Set<String> workgroupIdeas = getDetectedIdeas(workgroup, annotations);
+    return getWorkgroupsWithDifferentIdeas(possibleMembers, workgroupIdeas, annotations);
+  }
+
+  private Set<String> getDetectedIdeas(Workgroup workgroup, List<Annotation> annotations) {
+    Optional<Annotation> workgroupAnnotation = getWorkgroupAnnotation(workgroup, annotations);
+    return workgroupAnnotation.isPresent() ? getDetectedIdeas(workgroupAnnotation.get())
+      : new HashSet<String>();
+  }
+
+  private Set<String> getDetectedIdeas(Annotation annotation) {
+    try {
+      JSONObject dataJson = new JSONObject(annotation.getData());
+      JSONArray ideas = dataJson.getJSONArray("ideas");
+      Set<String> detectedIdeas = new HashSet<String>();
+      for (int i = 0; i < ideas.length(); i++) {
+        JSONObject idea = ideas.getJSONObject(i);
+        if (idea.getBoolean("detected")) {
+          detectedIdeas.add(idea.getString("name"));
+        }
+      }
+      return detectedIdeas;
+    } catch (JSONException e) {
+      return new HashSet<String>();
+    }
+  }
+
+  private TreeSet<WorkgroupWithDifferentIdeas> getWorkgroupsWithDifferentIdeas(
+      Set<Workgroup> possibleMembers, Set<String> workgroupIdeas, List<Annotation> annotations) {
+    TreeSet<WorkgroupWithDifferentIdeas> workgroups = new TreeSet<WorkgroupWithDifferentIdeas>();
+    for (Workgroup possibleMember : possibleMembers) {
+      if (hasWorkgroupAnnotation(possibleMember, annotations)) {
+        Set<String> possibleMemberIdeas = getDetectedIdeas(possibleMember, annotations);
+        Set<String> differentIdeas = SetUtils.disjunction(workgroupIdeas, possibleMemberIdeas);
+        workgroups.add(new WorkgroupWithDifferentIdeas(possibleMember, differentIdeas.size()));
+      }
+    }
+    return workgroups;
+  }
+}
```

Comment:
I noticed that the TreeSet can't contain duplicates which means for a given number of different ideas, it will only have one workgroup.

For example
Workgroup aa has idea 1 and is the one making the request to pair up
Workgroup bb has idea 2
Workgroup cc has ideas 3, 4
Workgroup dd has ideas 3, 4

The workgroups TreeSet will end up with only bb and cc. It will not contain dd because cc and dd have the same number of differentIdeas compared to aa.